wsmaster/codenvy-hosted-api-resource/src/main/java/com/codenvy/resource/api/free/FreeResourcesManager.java

```diff
@@ -0,0 +1,128 @@
+public class FreeResourcesManager {
+    private final FreeResourcesLimitDao   freeResourcesLimitDao;
+    private final ResourceInstanceCreator resourceInstanceCreator;
+
+    @Inject
+    public FreeResourcesManager(FreeResourcesLimitDao freeResourcesLimitDao,
+                                ResourceInstanceCreator resourceInstanceCreator) {
+        this.freeResourcesLimitDao = freeResourcesLimitDao;
+        this.resourceInstanceCreator = resourceInstanceCreator;
+    }
+
+    /**
+     * Stores (creates new one or updates existed) free resource limit.
+     *
+     * @param freeResourcesLimit
+     *         resources limit to store
+     * @return stored resources limit
+     * @throws NullPointerException
+     *         when {@code freeResourcesLimit} is null
+     * @throws NotFoundException
+     *         when resources limit contains resource with non supported type
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    public FreeResourcesLimit store(FreeResourcesLimit freeResourcesLimit) throws NotFoundException,
+                                                                                  ServerException {
+        requireNonNull(freeResourcesLimit, "Required non-null free resources limit");
+        List<AbstractResource> resources = new ArrayList<>();
+        for (Resource resource : freeResourcesLimit.getResources()) {
+            resources.add(resourceInstanceCreator.newInstance(resource));
+        }
+
+        final FreeResourcesLimitImpl toStore = new FreeResourcesLimitImpl(freeResourcesLimit.getAccountId(),
+                                                                          resources);
+        freeResourcesLimitDao.store(toStore);
+        return toStore;
+    }
+
+    /**
+     * Returns free resources limit for account with specified id
+     *
+     * @param accountId
+     *         account id to fetch resources limit
+     * @return free resources limit for account with specified id
+     * @throws NullPointerException
+     *         when {@code accountId} is null
+     * @throws NotFoundException
+     *         when free resources limit for specifies id was not found
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    public FreeResourcesLimit get(String accountId) throws NotFoundException, ServerException {
+        requireNonNull(accountId, "Required non-null account id");
+        return freeResourcesLimitDao.get(accountId);
+    }
+
+    /**
+     * Removes free resources limit for account with specified id
+     *
+     * Note: after removing resources limit account will be able to use default resources
+     *
+     * @param accountId
+     *         account id to remove resources limit
+     * @throws NullPointerException
+     *         when {@code accountId} is null
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    void remove(String accountId) throws ServerException {
```

Comment:
The common question here is what if i pass id of non-existing account, it seems that this method does nothing in this case(which i think is really good) but the things like that should be documented by method documentation.
BTW does it have to be package-private?